src/docs-developer-guide.rst

```diff
@@ -122,3 +122,188 @@ For example:
   }
 
   /* New section ends */
+
+.. _style-tests:
+
+Style Guide checks
+--------------------
+
+Proselint_  is used for style testing the docs.
+Apart from the inbuilt tests in proselint,
+custom checks are added for style guide testing.
+We follow the approach of `literate programming <https://en.wikipedia.org/wiki/Literate_programming>`_.
+The checks reside in the file :doc:`docs-style-guide.rst <docs-style-guide>`.
+After each style rule,
+you can define a python code-block
+containing the code for style testing.
+When style testing is performed,
+these python code-blocks are parsed to
+generate a testing script.
+
+.. _proselint-checks:
+
+Proselint dependent checks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Various inbuilt functions from Proselint are used to define the tests:
+
+.. py:function:: memoize
+
+  To cache the check for faster execution. Use :py:data:`@memoize` above function definition to cache the result.
+
+.. py:function:: existence_check(text, list, err, msg, ignore_case=True, \
+                                 str=False, max_errors=float("inf"), offset=0, \
+                                 require_padding=True, dotall=False, \
+                                 excluded_topics=None, join=False)
+
+  To check for existence of a regex pattern(s) in the text.
+  The parameters :py:data:`offset`, :py:data:`excluded_topics` and :py:data:`join`
+  are not needed for style guide testing.
+
```

Comment:
Does proselint really have function args with the names `list` and `str`?

Reply:
Yes, it actually does: https://github.com/amperser/proselint/blob/master/proselint/tools.py#L303

Reply:
Those are reserved words in Python.
Wacky.